New Away Mission: Snow Valley+snow atmos

MMMiracles opened the pull request:
:cl: MMMiracles
rscadd: A set of coordinates have recently reappeared on the Nanotrasen gateway project. Ask your local central official about participation.
/:cl:

You can't stop me.

A large away mission based around a valley coated in thick snow and ice, harboring multiple nanotrasen-held outposts for plot-related reasons. Insulation, a gun, and a few friends is key for survival in this harsh cold front.

There is a topside location where all players spawn with various outposts scattered across with small, static loot spots. Nothing too special, but for the risk required, it isn't bad loot. The good stuff comes on down lower, in the depths of the valley.

A large cave system is under the valley, harboring all kinds of undead nasties and loot. All loot spawns in from a set table of a lite, medium, and heavy table. This helps prevent singular strats to get the 'best' loot, as there is no guarantee the 'best' item will be in the same spot.

There's other stuff too but that'd ruin the fun of exploration so either go play it yourself or being a big fat weenie and look at the map in dreammaker or something.

Minimap of current progress (a lot):
![wew5](https://cloud.githubusercontent.com/assets/9276171/11648529/7d9a1c9a-9d43-11e5-8ccc-d8d8e6640db1.png)

The meat of it is done and most issues spotted during feedback testing were ironed out as far as I'm aware. At most maybe another feedback test for any minor tweaks before i'ts finally done.

MMMiracles committed:
start of something

MMMiracles committed:
foundation set, demands given.

AnturK commented:
There's already /plating/snow in misc_floor.dm , could you edit these instead ?

MMMiracles commented:
did not notice that woop

Anonus commented:
I hope you don't get cold feet trying to do this.

Jazaen commented:
Hm, would it be possible to resprite some mining mobs, creating winter versions?
Or just having some Basilisks to entertain the deadchat.

swankcookie commented:
ill take a shot at the respriting and post some protos

MMMiracles committed:
iced variant of regular walls added

MMMiracles committed:
shh

MMMiracles committed:
iced windows, more ice variants of shit in general.

more stuff, ice everything.

wew

conflict fix

Zilenan91 commented:
So are you making an Ice Caves part of the map? Normal basilisks should fit there if you are, they already have the color palette for it.

swankcookie commented:
@MMMiracles I finished the sprites and wrote the code for the new snow mobs. Its on my mining_mob_resprite branch if you want to check them out.  Also this: http://imgur.com/p211knK

MMMiracles commented:
The current sprites for the ice and rocks are placeholder and I'd very much appreciate if someone could sprite up something that fits more of the game's style.

@swankcookie I was looking for more 'original' ice mobs other than resprites of the mining mobs, but for now they'll do until then.

swankcookie commented:
@MMMiracles your welcome.

MMMiracles committed:
snow walls+material

MMMiracles committed:
snow stuff.

MMMiracles committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into snowiscold

Conflicts:
	icons/mob/animal.dmi

MMMiracles commented:
i did not realize how much stuff i'd need to make beforehand to actually have content to work with.

still can't stop me.

Jordie0608 commented:
You'll need to add this to the away mission configs too.

MMMiracles committed:
finalization of the topside part, time to make dungeon designs woo

MMMiracles commented:
I'm gonna spend a bit thinking out designs for the dungeons but if anyone has any ideas please send me a forum PM or something because I honestly ain't got much in terms of ideas to work with.

Zilenan91 commented:
Who are you on the forums?

MMMiracles commented:
Take a wild guess.

Zilenan91 commented:
I can't find MMMiracles under M by searching for users on the forums

MMMiracles commented:
Thats literally my name so thats weird.

Zilenan91 commented:
Actually wait, this fucking member list isn't alphabetized god damnit

Zilenan91 commented:
Oh hey quick question what kinda of enemies are outside the dungeon itself so I can maybe theme it

MMMiracles commented:
The only real threat is the local wildlife outside of the dungeon 'polar bears' and maybe a skeleton or two.

Zilenan91 commented:
Is there any kind of settlement there or anything?

MMMiracles commented:
Other than outposts, no.

Zilenan91 commented:
What are in the outposts? Any syndicates?

MMMiracles commented:
nop no syndies

Zilenan91 commented:
Do you already have the exit gateway mapped somewhere or would it be in the dungeon

MMMiracles commented:
Gateway is already mapped topside.

Zilenan91 commented:
So what mobs are inside of the outposts?

MMMiracles committed:
wew

MMMiracles commented:
topside isnt really meant to be combat-based, the dungeons are, so there isnt really any mobs in the outposts.

Zilenan91 commented:
Woops, forgot to edit a part in, the syndicate channel would be usable on this away mission so traitors can activate and then ally with the syndies there. If you want to avoid traitors buddying up with away mission antags consider having it in their objectives to punish all syndicate agents who are disobeying the Syndicate and fucking off to the away mission.

MMMiracles committed:
fianl stuff for now, testing!!!!!

MMMiracles committed:
woops

MMMiracles commented:
gonna start looking for feedback now, hopefully on-server or something.

xxalpha commented on code/game/objects/structures/window.dm:
```diff
@@ -483,6 +483,15 @@
+	dir = 5
+	maxhealth = 150
+	fulltile = 1
+	smooth = SMOOTH_TRUE
```
Don't need to set this, the parent type is already SMOOTH_TRUE.

MMMiracles replied:
Apparently not, the other subtypes do this and not doing this causes it to not smooth.

xxalpha replied:
Vars from /obj/structure/window/reinforced/fulltile are inherited by /obj/structure/window/reinforced/fulltile/ice

xxalpha commented on code/game/turfs/simulated/walls_mineral.dm:
```diff
@@ -162,4 +162,15 @@
+	mineral = "snow"
+	hardness = 80
+	sheet_type = /obj/item/stack/sheet/mineral/snow
+	canSmoothWith = list(/turf/simulated/wall/mineral/snow)
```
To smooth with the exact same type, leave this var null.

MMMiracles replied:
Doesn't work, other subtypes do the same thing.

xxalpha commented on code/game/turfs/simulated/walls_misc.dm:
```diff
@@ -26,6 +26,15 @@
+	icon_state = "iced"
+	desc = "A wall covered in a thick sheet of ice."
+	walltype = "iced"
+	canSmoothWith = list(/turf/simulated/wall/ice)
```
To smooth with only others of the same type, leave this var null.

MMMiracles replied:
Doesn't work, other subtypes do the same thing.

xxalpha replied:
I can guarantee you it works.

MMMiracles replied:
I dunno man, I was literally just doing a local test without it and they only smoothed with the regular walls and not themselves.

xxalpha replied:
Set it to null.

xxalpha commented:
The riced_windows.dmi is not properly made, all the icons are bottom left corners. You done goofed.

MMMiracles commented:
Yeah I saw that last night, already fixed it locally.

MMMiracles committed:
more map edits, loot table balancing, bug fixes, also ATV VROOM VROOM

MMMiracles committed:
more map edits, loot table balancing, bug fixes, also ATV VROOM VROOM

MMMiracles committed:
woops

MMMiracles committed:
Merge branch 'snowiscold' of https://github.com/MMMiracles/-tg-station into snowiscold

WJohn commented:
Big :-1: on the ATV sprite. It needs significantly darker outlines than the fills, clear differences between each shade, highlights, and would probably need to be a different color than white entirely because you'll already be surrounded by white. On the upside, you kept the flat-in-front style correct, though I'm not sure if that's supposed to be used for vehicles or not. That's not really established anywhere.

MMMiracles commented:
I was actually going to try and make the top segment a grayscaled overlay so there could be colored variants of it. I'm not exactly the best spriter so help would be appreciated.

MMMiracles committed:
mostly final changes, wowzers

MMMiracles committed:
thats not suppose to be there

MMMiracles committed:
oh my

MMMiracles commented:
I've made my changes, operatives should no longer be able to easily access their armory of high-damage goodies without spending like 10-15 minutes welding through r-walls.

Its had 2 feedback tests so I think its past the review stage and its ready for merge.

MMMiracles committed:
conflict fix

MMMiracles commented:
no conflicts here officer

MMMiracles committed:
lets try that again

MMMiracles commented:
merge y/y

MMMiracles commented:
?????

optimumtact commented:
@Aranclanos @Cheridan @Jordie0608 @phil235 @WJohn @KorPhaeron @caelaislinn @duncathan @Razharas @swankcookie @neersighted @MMMiracles @AndrewJacksonThe2nd @AnturK @Anonus @SuperSayu @RemieRichards @octareenroon91 @Yolopanther12 @hornygranny @YotaXP @Bawhoppen @bgobandit @MrStonedOne 

MERGE THIS ALREADY WHAT ARE YOU DOING

optimumtact commented:
@Iamgoofball

swankcookie commented:
@optimumtact

SuperSayu commented:
what am I a maintainer now

fine then time to do all sorts of crazy shit

KazeEspada commented on code/game/objects/items/toys.dm:
```diff
@@ -1169,6 +1170,26 @@
+	desc = "A compact ball of snow. Good for throwing at people."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "snowball"
+	throwforce = 12 //pelt your enemies to death with lumps of snow
```
Thats a lot of force for a ball of snow.

MMMiracles replied:
Someone getting pelted to death by snowballs seems funny to me, its just really hard snow.

KazeEspada replied:
Thats more damage than hitting someone with a fire extinguisher.

octareenroon91 replied:
Very severe for a toy.
```damtype = STAMINA```

MMMiracles replied:
Its not meant to be a real 'toy', unless being inside the toys.dm is enough to qualify it for an arcade toy prize. I couldn't think of any other decent spot for it and they're hard to get anyway.

tkdrg replied:
well, christmas gifts can pick from these since they pick a random toy
sounds like a feature to me though

swankcookie replied:
It kinda fits with christmas presents, And it will be hilarious to see someone accidentally fuck someone up with it.

KazeEspada commented on code/game/objects/structures/janicart.dm:
```diff
@@ -325,4 +329,45 @@
+
+//atv//
+
+/obj/structure/bed/chair/janicart/atv
```
I think this typepath is a bit out of control.

MMMiracles replied:
That's how the secway does it so I dunno.

KorPhaeron replied:
Janicarts are beds because they used to be the only objects you could buckle to

KazeEspada replied:
I didnt mean he had to fix it, I was just commenting that this type path is only gonna get uglier.

swankcookie replied:
yeah it's especially weird considering the newer janitorial_cart is 
/obj/structure/janitorialcart
there is no logical reason for jancart to be that way

duncathan commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -237,6 +238,32 @@ var/global/list/datum/stack_recipe/clown_recipes = list ( \
+	new/datum/stack_recipe("Snow Wall",/turf/simulated/wall/mineral/snow, 5, one_per_turf = 1, on_floor = 1), \
+	new/datum/stack_recipe("Snowman", /obj/structure/statue/snow/snowman, 5, one_per_turf = 1, on_floor = 1), \
+	new/datum/stack_recipe("Snowball", /obj/item/toy/snowball, 1), \
+	)
```
Please tell me this isn't standard cuz this looks bad

KorPhaeron replied:
It's standard yeah. Construction sucks

duncathan commented on code/game/objects/structures/statues.dm:
```diff
@@ -350,3 +350,14 @@
+
+/obj/structure/statue/snow
+	hardness = 0.5
+	mineralType = "sandstone"
```
?

MMMiracles replied:
jokes on you the entire mission is actually just white sand

in other words copypasta is bad

duncathan commented on code/game/turfs/simulated/floor/misc_floor.dm:
```diff
@@ -45,17 +45,65 @@
+	return
+
+/turf/simulated/floor/plating/snow/burn_tile()
+	return
```
>can't burn snow

MMMiracles replied:
no good melted-snow type of sprite I could use so its just really cold snow.

duncathan commented on code/game/turfs/simulated/floor/misc_floor.dm:
```diff
@@ -45,17 +45,65 @@
+		if(do_after(user, digging_speed, target = src))
+			if(istype(src, /turf/simulated/floor/plating/snow))
+				user << "<span class='notice'>You shovel a path in the snow.</span>"
+				gets_dug()
```
This has to be copy paste

MMMiracles replied:
it is, if theres a less copy-pasta way please let me know

optimumtact replied:
complaining about this is silly, there's two types of digging right now, asking an away mission maker to generalise a digging system is silly.

KorPhaeron replied:
They could make snow a child of asteroid turfs I guess?

And add a var for what kind of "sand' gets dug up.

duncathan replied:
Yeah that's what I was thinking @KorPhaeron

MMMiracles replied:
if i make snow a child of asteroid turfs then i'd have to edit the wizard and centcomm zlevels.

pls.

KorPhaeron replied:
Don't change the fake snow on those levels then. That can be done later.

MMMiracles replied:
Then there'd have to be two different turfs of snow that only difference is one is diggable.

KorPhaeron replied:
That is fine for now, we have fake walls and windows and doors on centcomm as well.

duncathan commented on code/game/turfs/simulated/floor/misc_floor.dm:
```diff
@@ -63,6 +111,21 @@
+	icon_state = "ice"
+	baseturf = /turf/simulated/floor/plating/snow
+	slowdown = 1
+	wet = 3
```
Don't you have a define for this?

duncathan commented on code/modules/awaymissions/mission_code/snowdin.dm:
```diff
@@ -0,0 +1,281 @@
+
+//special items//--
+
+/obj/item/clothing/belt/
```
?

MMMiracles replied:
i think that was a loot item i was thinking of making but didn't bother finishing, guess i forgot to remove the path

duncathan commented on code/modules/awaymissions/mission_code/snowdin.dm:
```diff
@@ -0,0 +1,281 @@
+	desc = "This wand shoots scorching balls of fire that explode into destructive flames. The years of the cold have weakened the magic inside the wand."
+	max_charges = 4
+
+/obj/item/weapon/gun/magic/wand/resurrection/innert
```
inert

duncathan commented on code/modules/mob/living/simple_animal/hostile/skeleton.dm:
```diff
@@ -28,11 +31,79 @@
+		new loot (src.loc)
+	if(loot2)
+		new loot2 (src.loc)
+	if(loot3)
```
Make loot a list

duncathan commented on code/modules/mob/living/simple_animal/hostile/skeleton.dm:
```diff
@@ -28,11 +31,79 @@
+	loot3 = /obj/item/weapon/claymore/hog{name = "holy sword"}
+
+/mob/living/simple_animal/hostile/skeleton/templar/bullet_act(obj/item/projectile/Proj)
+	if(!Proj)	return
```
No same line returns

MMMiracles committed:
i did things

MMMiracles committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into snowiscold

MMMiracles committed:
snow stuff i guess

MMMiracles commented:
I've addressed the comments from earlier for the most part, the snow used on the mission is a subtype of the asteroid turf and I left the regular plating snow alone because I don't want to add two more maps edited to this PR.

MMMiracles committed:
woops

tkdrg commented:
this doesn't compile bro
travis tells no lies
`code/game/turfs/simulated/floor/misc_floor.dm:63:error: dug: undefined var`

MMMiracles commented:
cant wake up

MMMiracles committed:
wake me up

MMMiracles committed:
conFLICTS

tkdrg commented on code/game/objects/items/toys.dm:
```diff
@@ -1169,6 +1170,26 @@
+
+/obj/item/toy/snowball/afterattack(atom/target as mob|obj|turf|area, mob/user)
+	user.drop_item()
+	src.throw_at(target, throw_range, throw_speed)
```
don't need the src here but it's not that big of a deal

MMMiracles replied:
I took influence from how the beach ball handles throwing stuff so blame it.

tkdrg commented on code/game/turfs/simulated.dm:
```diff
@@ -51,7 +51,16 @@
 	smooth_icon_neighbors(src)
+
+turf/simulated/MakeSlippery()
+	if(TURF_WET_ICE)
```
You sure that this is what you want to do? This is just `if(3)` which is always true.

tkdrg commented:
It would also be nice if you could squash, this doesn't need 27 commits.

MMMiracles commented:
Last guide I tried on squashing commits messed up differences between local and remote and had me restart on a new folder.

A simplified guide on that would be helpful.

tkdrg commented:
There's not really much to write a guide on for it.
`git rebase -i HEAD~28` to enter an interactive rebase, mark all the later commits as squash, and reword the first one to a more adequate commit message. And if you fuck up, you just gotta ask oranges to send help on #coderbus.

tkdrg commented:
Also don't forget to fix `MakeSlippery()`, that line is literally broken.

MMMiracles committed:
nvm

tkdrg commented:
Are you going to try to squash?

MMMiracles commented:
the command brings up nothing in gitbash and trying it in tortoisegit is plagued with conflicts.

how do i computer.

tkdrg commented:
eh whatever, our commit log is absolute shit already
